components/ai_chat/renderer/page_content_extractor.cc

```diff
@@ -318,4 +321,54 @@ void PageContentExtractor::GetSearchSummarizerKey(
   std::move(callback).Run(element.GetAttribute("content").Utf8());
 }
 
+void PageContentExtractor::ValidateOpenLeoButtonNonce(
+    const std::string& nonce,
+    mojom::PageContentExtractor::ValidateOpenLeoButtonNonceCallback callback) {
+  if (nonce.empty()) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  auto element = render_frame()->GetWebFrame()->GetDocument().GetElementById(
+      "continue-with-leo");
+  if (element.IsNull() || !element.HasHTMLTagName("a")) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  GURL url(element.GetAttribute("href").Utf8());
+  if (!IsOpenLeoButtonFromBraveSearchURL(url) ||
+      !element.HasAttribute("nonce") || url.ref_piece() != nonce) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  // Value of nonce property is not accessible via GetAttribute API, so we need
```

Comment:
does this mean if we renamed it to a different attribute, we could get it via the GetAttribute API and not have to execute a script?

Reply:
Asked a search-dev to help test what happens if we put in data-nonce.

Reply:
great, it would be nice to get rid of this script injection

Reply:
why are we doing this in the tag instead of in the url? This seems to complicate things. Also why do we need two methods to get this? If we just used JS we wouldn't need anything on the renderer side at all. This stuff also doesn't seem like it belongs in `PageContentExtractor`

Reply:
also couldn't we get this from axtree if it's an attribute of a tag?

Reply:
> why are we doing this in the tag instead of in the url? 

the point of this security check is to prevent an attacker who can get links into the brave search webpage from being able to open leo once the user clicks that link. in order to do such an attack, an attacker has to both be able to insert links into search AND either know this random nonce (which is generated every time the page loads) or be able to modify HTML attributes on search.

example scenarios: 
1. attacker runs a search ad for search.brave.com/leo and this somehow gets through our ads validation checks
2. search adds a feature where goggles authors can insert a custom "learn more" link

the point is that the search server should not be solely responsible for making sure nothing except the "open in leo" button initiates a navigation from search.brave.com to search.brave.com/leo

Reply:
FYI I'm pushing a fix to use data-nonce, we won't have script execution code here anymore, would just be element.GetAttribute("data-nonce").

Reply:
changed to data-nonce in 52a60130f4680ac6b22955206ff08c0ad47d127a